rust/lance/src/dataset/write.rs

```diff
@@ -80,6 +81,15 @@ pub struct WriteParams {
     pub store_params: Option<ObjectStoreParams>,
 
     pub progress: Arc<dyn WriteFragmentProgress>,
+
+    /// If present, dataset will use this to update the latest version
+    ///
+    /// If not set, the default will be based on the object store.  Generally this will
+    /// be RenameCommitHandler unless the object store does not handle atomic renames (e.g. S3)
+    ///
+    /// If a custom object store is provided (via store_params.object_store) then this
+    /// must also be provided.
+    pub commit_handler: Option<Arc<dyn CommitHandler>>,
```

Comment:
Should we just provide a default file path based CommitHandler, from the caller side we dont need to do

```
match commit_handler  {
   Some(ch) => {. ch.commit() }
   None => { commit_other_way() }
}
```

Reply:
I'm not sure I understand.  We do have a default commit handler based on the URL.  That's why this property is `Option<...>` since you can specify `None` to use the default.